### PR TITLE
[refactor] BF-202 Delete unused code and refactor project according to best practicies

### PR DESCRIPTION
**Pull request by korzdan**

### CODE REVIEW

File: src/main/java/by/bookify/auth/controller/AuthenticationController.java [hunk 1]
```diff
@@ -12,26 +12,29 @@
...
     public ResponseEntity<AuthResponseDto> registerUser(@RequestBody RegisterRequestDto registerRequestDto) {
-        return ResponseEntity.ok(authService.registerUser(registerRequestDto));
+        return ResponseEntity.ok(authService.registerUser(registerRequestDto, USER));
     }
 
     @PostMapping("/register/admin")
     @PreAuthorize("hasRole('SUPER_ADMIN')")
     public ResponseEntity<AuthResponseDto> registerAdmin(@RequestBody RegisterRequestDto registerRequestDto) {
-        return ResponseEntity.ok(authService.registerAdmin(registerRequestDto));
+        return ResponseEntity.ok(authService.registerUser(registerRequestDto, ADMIN));
```
Comment (KaShTaN27) [inline src/main/java/by/bookify/auth/controller/AuthenticationController.java]: Renaming proposal to register(). Cause registerUser with role Admin can be confusing

File: src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class GenreExceptionHandler {
+
+    @ExceptionHandler(GenreNotFound.class)
+    public ResponseEntity<ExceptionResponse> handleGenreNotFound(GenreNotFound e) {
+        return new ResponseEntity<>(new ExceptionResponse(e.getMessage()), HttpStatus.NOT_FOUND);
```
Comment (KaShTaN27) [inline src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java]: Code style issue. In controllers you are using static methods of ResponseEntity, but in exception handlers you instantiate through constructor

File: src/main/java/by/bookify/auth/controller/AuthenticationController.java (hunk 1 above)
Comment (korzdan) [reply src/main/java/by/bookify/auth/controller/AuthenticationController.java]: Exacltly.
Fixed

File: src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java (hunk 2 above)
Comment (korzdan) [reply src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java]: The others adviceControllers use different codes where there are no statics methods for some codes. What to do?

Comment (KaShTaN27) [reply src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java]: ResponseEntity.status()

File: src/main/java/by/bookify/book/repository/BookRepository.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package by.bookify.book.repository;
+
+import by.bookify.book.model.Book;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.List;
+
+public interface BookRepository extends CrudRepository<Book, Long> {
```
Comment (KaShTaN27) [inline src/main/java/by/bookify/book/repository/BookRepository.java]: Why CrudRepository and not a jpa one. With crud you are not be able to paging

File: src/main/java/by/bookify/user/repository/UserRepository.java [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+import by.bookify.user.model.User;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 import java.util.Optional;
 
 public interface UserRepository extends JpaRepository<User, Long> {
 
     Optional<User> findByEmail(String email);
+
     Boolean existsByEmail(String email);
```
Comment (KaShTaN27) [inline src/main/java/by/bookify/user/repository/UserRepository.java]: Here you can return primitive and not to create an object

**Review by KaShTaN27 — CHANGES_REQUESTED** [review]

File: src/main/java/by/bookify/order/service/DefaultOrderService.java [hunk 5]
```diff
@@ -37,17 +40,34 @@ public List<Order> findByUserId(Long userId) {
     @Override
     public List<Order> findByStatus(String status) {
         try {
-            OrderStatus orderStatus = OrderStatus.valueOf(status);
-            return orderRepository.findByStatus(orderStatus);
+            return orderRepository.findByStatus(OrderStatus.valueOf(status));
         } catch (IllegalArgumentException e) {
             throw new OrderNotFound("Такого статуса не существует.");
```
Comment (antonydanon) [inline src/main/java/by/bookify/order/service/DefaultOrderService.java]: You translated everything into English

Comment (antonydanon) [reply src/main/java/by/bookify/order/service/DefaultOrderService.java]: Thrown exception is not suitable for this method

File: src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java (hunk 2 above)
Comment (korzdan) [reply src/main/java/by/bookify/genre/exception/GenreExceptionHandler.java]: Done

File: src/main/java/by/bookify/book/repository/BookRepository.java (hunk 3 above)
Comment (korzdan) [reply src/main/java/by/bookify/book/repository/BookRepository.java]: Done

File: src/main/java/by/bookify/user/repository/UserRepository.java (hunk 4 above)
Comment (korzdan) [reply src/main/java/by/bookify/user/repository/UserRepository.java]: Done

File: src/main/java/by/bookify/order/service/DefaultOrderService.java (hunk 5 above)
Comment (korzdan) [reply src/main/java/by/bookify/order/service/DefaultOrderService.java]: Translated, throw sutable exception

**Review by KaShTaN27 — APPROVED** [review]

**Review by antonydanon — APPROVED** [review]

**Pull request merged**